### ISSUE TITLE
DFS: Depth First Search does not exist

### ISSUE THREAD

**Issue by michaelasapp** [igraph/python-igraph#295]
*Problem*: I do not see that `graph.dfs` is exposed in the Python API to the C++ library.

I see commit b0d74bd54224a3c34febf3e0444f4198a93772d0 that adds BFS and the title include DFS, but the code does not refer to DFS.

**Comment by rcalfredson** [comment]
Coincidentally, just ran into the same problem.

**Comment by michaelasapp** [comment]
I should point out it's certainly not a big deal -- I spent 10 minutes searching the docs, 5 minutes complaining, and 5 minutes writing a small function -- but it was surprising when porting code from R to Python.

FWIW @rcalfredson I just used this.

```python
def dfs(graph, node):
    yield node
    for edge in graph.vs[node].out_edges():
        yield from dfs(graph, edge.target)
```

**Comment by ntamas** [comment]
Sorry guys, will fix this soon (during the weekend if everything goes well).

**Comment by michaelasapp** [comment]
Thank you!

**Comment by ntamas** [comment]
This is taking longer than expected; I was under the impression that `Graph.bfs()` is based directly on `igraph_bfs()` so I could just copy the implementation and replace a few arguments here and there, but it turns out that `Graph.bfs()` is using `igraph_i_bfs()`, an internal and simplified counterpart of `igraph_bfs()` so the code cannot simply be copied.

**Comment by iosonofabio** [comment]
closed via #315
